core/src/sumeragi/message.rs

```diff
@@ -470,47 +151,6 @@ pub struct BlockCommitted {
     pub block: VersionedValidBlock,
 }
 
-impl BlockCommitted {
-    /// Handles this message as part of `Sumeragi` consensus.
-    #[iroha_futures::telemetry_future]
-    pub async fn handle<G: GenesisNetworkTrait, F: FaultInjection>(
-        &self,
-        sumeragi: &mut SumeragiWithFault<G, F>,
-    ) -> Result<()> {
-        let network_topology = sumeragi.network_topology_current_or_genesis(self.block.header());
-        #[allow(clippy::expect_used)]
-        let network_topology = network_topology
-            .into_builder()
-            .with_view_changes(self.block.header().view_change_proofs.clone())
-            .build()
-            .expect("When only changing view changes it should not fail.");
-        let verified_signatures = self
-            .block
-            .verified_signatures()
-            .cloned()
-            .collect::<Vec<_>>();
-        let valid_signatures = network_topology.filter_signatures_by_roles(
-            &[Role::ValidatingPeer, Role::Leader, Role::ProxyTail],
-            &verified_signatures,
-        );
-        let proxy_tail_signatures =
-            network_topology.filter_signatures_by_roles(&[Role::ProxyTail], &verified_signatures);
-        if valid_signatures.len() >= network_topology.min_votes_for_commit()
-            && proxy_tail_signatures.len() == 1
-            && sumeragi.latest_block_hash() == &self.block.header().previous_block_hash
-        {
-            let mut block = self.block.clone();
```

Comment:
Did you just move it to the Sumeragi it-self? Am I right?

Reply:
Yes